lib/src/views/home/widgets/recipe_list.dart

```diff
@@ -0,0 +1,298 @@
+        });
+      } else {
+        setState(() {
+          isLoading = false;
+          recipes = [];
+        });
+      }
+    } catch (e) {
+      // Run after widget has been rendered and built
+      WidgetsBinding.instance.addPostFrameCallback(
+          (_) => ScaffoldMessenger.of(context).showSnackBar(
+                SnackBar(
+                  content: Text('An error occurred while loading the data: $e'),
+                  backgroundColor: Colors.red,
+                ),
+              ));
+      setState(() {
+        isLoading = false;
+        recipes = [];
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _buildContent();
+  }
+
+  Widget _buildContent() {
+    if (isLoading) {
+      return _buildLoadingList();
+    } else if (recipes.isNotEmpty) {
+      return _buildLoadedList();
+    } else {
+      return _buildEmptyList();
+    }
+  }
+
+  Center _buildEmptyList() {
+    return Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          const Icon(Icons.error_outline, size: 48, color: Colors.red),
+          const SizedBox(height: 16),
+          const Text('No recipes found', style: TextStyle(fontSize: 18)),
+          const SizedBox(height: 16),
+          ElevatedButton(
+            onPressed: fetchData,
+            child: const Text('Retry'),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildLoadedList() {
+    return RefreshIndicator(
+      onRefresh: fetchData,
+      child: OrientationBuilder(
+        builder: (context, orientation) {
+          return GridView.builder(
+            gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                mainAxisExtent: orientation == Orientation.portrait
+                    ? context.screenSize.width * 0.35
+                    : context.screenSize.height * 0.75,
+                crossAxisCount: orientation == Orientation.portrait ? 1 : 3,
+                mainAxisSpacing: 16,
+                crossAxisSpacing: 16),
+            itemCount: recipes.length,
+            padding: EdgeInsets.only(
+              left: orientation == Orientation.landscape
+                  ? MediaQuery.of(context).padding.left
+                  : 16,
+              right: orientation == Orientation.landscape
+                  ? MediaQuery.of(context).padding.right
+                  : 16,
+              bottom: MediaQuery.of(context).padding.bottom,
+            ),
+            itemBuilder: (context, index) {
+              return GestureDetector(
+                onTap: () {
+                  Navigator.push(context, MaterialPageRoute(builder: (context) {
+                    return RecipeDetail(recipe: recipes[index]);
+                  }));
+                },
+                child: RecipeCard(
+                  recipe: recipes[index],
+                ),
+              );
+            },
+          );
+        },
+      ),
+    );
+  }
```

Comment:
It's best to define a `final` variable in case the specific value is use more than 1 time.

```suggestion
  Widget _buildLoadedList() {
    return RefreshIndicator(
      onRefresh: fetchData,
      child: OrientationBuilder(
        builder: (context, orientation) {
          final padding = MediaQuery.paddingOf(context);
          final isLandscape = orientation == Orientation.landscape;
          final screenSize = context.screenSize;

          return GridView.builder(
            gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                mainAxisExtent: !isLandscape
                    ? screenSize.width * 0.35
                    : screenSize.height * 0.75,
                crossAxisCount: !isLandscape ? 1 : 3,
                mainAxisSpacing: 16,
                crossAxisSpacing: 16),
            itemCount: recipes.length,
            padding: EdgeInsets.only(
              left: isLandscape ? padding.left : 16,
              right: isLandscape ? padding.right : 16,
              bottom: padding.bottom,
            ),
            itemBuilder: (context, index) {
              return GestureDetector(
                onTap: () {
                  Navigator.push(context, MaterialPageRoute(builder: (context) {
                    return RecipeDetail(recipe: recipes[index]);
                  }));
                },
                child: RecipeCard(
                  recipe: recipes[index],
                ),
              );
            },
          );
        },
      ),
    );
  }
```